### PR TITLE
Players - WIP, could use some review

### PR DESCRIPTION
**Pull request by lindyrp**
My commit comments are pretty clear in what is covered (I think).

There are two main things I'd still like to do before pulling in the whole branch that are NOT yet done here:
* moving "Create Game" and "Join Existing Game" into modals directly from the navbar
* want to add more controller tests

Let me know your questions/thoughts so far. Thanks!

### CODE REVIEW

**Comment by ruishine** [comment]
above and beyond! Thank you, Lindy! will take a closer look this evening

On Mon, Jul 11, 2016 at 2:22 PM, Lindy <notifications@github.com> wrote:

> My commit comments are pretty clear in what is covered (I think).
>
> There are two main things I'd still like to do before pulling in the whole
> branch that are NOT yet done here:
>
>    - moving "Create Game" and "Join Existing Game" into modals directly
>    from the navbar
>    - want to add more controller tests
>
> Let me know your questions/thoughts so far. Thanks!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Methodologists/metchess/pull/27
> Commit Summary
>
>    - Start branch and add foreign key
>    - Add name field to game, separate new and index views
>    - White player set to current user when creating new game
>    - Index lists only games without second player
>    - Starting tests for creating games
>    - Resolve Gemfile.lock conflict
>    - Add indices for player_white and player_black ids to games table
>    - Add sign in/out links and form to join existing game
>    - Add top navbar to home page, hook up create and join actions
>    - Minor styling
>    - Display white player email for join game index
>    - Hook up My Games show page
>    - Show My Games link only if user signed in
>    - My Games link shows all games for current user by player color
>
> File Changes
>
>    - *M* Gemfile
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-0> (2)
>    - *M* Gemfile.lock
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-1> (4)
>    - *A* app/assets/stylesheets/master.css.scss
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-2> (4)
>    - *M* app/controllers/games_controller.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-3> (8)
>    - *M* app/controllers/users_controller.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-4> (5)
>    - *M* app/models/game.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-5> (3)
>    - *M* app/models/user.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-6> (5)
>    - *M* app/views/games/index.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-7> (18)
>    - *A* app/views/games/new.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-8> (8)
>    - *M* app/views/games/show.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-9> (4)
>    - *M* app/views/layouts/application.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-10>
>    (55)
>    - *A* app/views/users/show.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-11>
>    (25)
>    - *A* config/initializers/simple_form.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-12>
>    (165)
>    - *A* config/initializers/simple_form_bootstrap.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-13>
>    (149)
>    - *A* config/locales/simple_form.en.yml
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-14>
>    (31)
>    - *M* config/routes.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-15> (1)
>    - *A* db/migrate/20160706175845_alter_games_add_name.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-16> (5)
>    - *A* db/migrate/20160708184450_alter_games_add_player_indices.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-17> (6)
>    - *A* db/migrate/20160711180721_alter_pieces_add_user_id.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-18> (5)
>    - *M* db/schema.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-19> (7)
>    - *A* lib/templates/erb/scaffold/_form.html.erb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-20>
>    (13)
>    - *M* spec/controllers/games_controller_spec.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-21>
>    (25)
>    - *M* spec/factories.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-22>
>    (10)
>    - *M* spec/rails_helper.rb
>    <https://github.com/Methodologists/metchess/pull/27/files#diff-23> (7)
>
> Patch Links:
>
>    - https://github.com/Methodologists/metchess/pull/27.patch
>    - https://github.com/Methodologists/metchess/pull/27.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Methodologists/metchess/pull/27>, or mute the thread
> <https://github.com/notifications/unsubscribe/ALOK7Fra9jFOb9z6vOKwYfajsaBLDgCSks5qUpfpgaJpZM4JJsU1>
> .
>



-- 
-Lin Rui Li

File: app/models/user.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class User < ActiveRecord::Base
-  has_many :games
-  has_many :pieces
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :games
+  has_many :pieces
```
Comment (ilyakrasnov) [inline app/models/user.rb]: Any use for that? Pieces are already associated with the game, right?

File: spec/controllers/games_controller_spec.rb [hunk 2]
```diff
@@ -2,4 +2,29 @@
...
+      game = Game.last
+      expect(game.name).to eq("Testgame")
+    end
+
+    it "should add the current user as white player when creating game" do
+      user = FactoryGirl.create(:user)
+      sign_in user
+
+      post :create, game: {name: 'Testgame',
+                            player_white_id: user.id}
```
Comment (ilyakrasnov) [inline spec/controllers/games_controller_spec.rb]: What happens if you don't specify 'player_white_id'. This is the exact thing you want to test that when creating a game the current user is automatically the white player..

**Comment by ilyakrasnov** [comment]
Looks very good and clean. Watch out for two comments.. Otherwise 👍

File: app/models/user.rb (hunk 1 above)
Comment (lindyrp) [inline app/models/user.rb]: You know, I thought the same thing - this is a remnant from the initial setup, I just moved its location in the code. It's echoed in `piece.rb`
```
class Piece < ActiveRecord::Base
  belongs_to :user
  belongs_to :game
```

File: spec/controllers/games_controller_spec.rb (hunk 2 above)
Comment (lindyrp) [inline spec/controllers/games_controller_spec.rb]: @ilyakrasnov: I know this one is a problem. I'm having a hard time formulating controller tests with the information being passed via forms. Could use some guidance here, and it's also on the agenda to discuss more thoroughly with my mentor midweek.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,2 +1,7 @@
 class UsersController < ApplicationController
+
+  def show
+    @games = Game.all
```
Comment (AlekstheRealist) [inline app/controllers/users_controller.rb]: Shouldn't we have this index in one place? We have the same functionality of `@games = Game.all` in the games_controller under `def index`

Comment (lindyrp) [inline app/controllers/users_controller.rb]: Unfortunately we don't. Without it here, we get the following error:
```
NoMethodError in Users#show
Showing /vagrant/src/metchess/app/views/users/show.html.erb where line #9 raised:

undefined method `each' for nil:NilClass
Extracted source (around line #9):
     <strong>As White Player:</strong>
     <br />
     <% @games.each do |game| %>
```
I've been trying to figure out if there's something wrong in the associations that is causing this, but adding it here fixed it, so I went with it. :)

**Pull request merged**